### PR TITLE
Android script + BUILD.md cleanup

### PR DESCRIPTION
**Pull request by juan-lunarg**

### CODE REVIEW

File: scripts/android.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import argparse
+import subprocess
+import sys
+import common_ci
+
+# TODO: Use CMake from Android SDK?
+
+# Android APKs can contain binaries for multiple ABIs (armeabi-v7a, arm64-v8a, x86, x86_64).
+# CMake will need to be run multiple times to create a complete test APK that can run on any Android device.
+def main():
```
Comment (juan-lunarg) [inline scripts/android.py:27]: Note. This script is pretty rough right now but I am able to successfully create Android binaries with CMake and create the test `VulkanLayerValidationTests.apk`

image

File: scripts/android.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+    print(f"ANDROID_NDK_HOME = {android_ndk_home}")
+    print(f"ANDROID_SDK_ROOT = {android_sdk_root}")
+    print(f'Android cmake toolchain file = {android_toolchain}')
+
+    # TODO: Verify aapt, ndk-build, cmake, ninja are in the path.
+
+    # TODO: Allow user to specify abis to save time.
+    # android_abis = ["armeabi-v7a", "arm64-v8a", "x86", "x86_64"]
+    android_abis = [ "armeabi-v7a", "arm64-v8a" ]
```
Comment (spencer-lunarg) [inline scripts/android.py:46]: honestly, I only ever build `arm64-v8a` locally, 32-bit is something, like `x86` I think people are opting into now

File: scripts/android.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    # Align APK
+    common_ci.RunShellCmd(f'zipalign -f 4 {unaligned_apk} {aligned_apk}')
+
+    
+
+    # If zipalign was run after signing, it won't be a valid signature
+
+    # TODO: Figure out how to sign APK. Looks like jarsigner is deprecated.
+    #
+    # jarsigner used to be used until it was removed from the Android SDK
```
Comment (spencer-lunarg) [inline scripts/android.py:119]: this seems left in by accident

**Review by spencer-lunarg — APPROVED** [review]
few questions, but looks good to me

File: scripts/android.py (hunk 2 above)
Comment (juan-lunarg) [reply scripts/android.py:46]: I currently am just making sure the APK has multiple binaries. I just picked the 2 arms ABIs arbitrarily for now to save time compiling.

File: scripts/android.py (hunk 3 above)
Comment (juan-lunarg) [reply scripts/android.py]: I'm going to do a follow up with the signing logic. I want this script to be flexible enough for vvl and jenkins.

File: scripts/android.py (hunk 2 above)
Comment (juan-lunarg) [reply scripts/android.py:46]: IE by default I think they should all be on. However, I want to give users the ability to specify how many they want in another PR.

File: scripts/android.py (hunk 3 above)
Comment (juan-lunarg) [reply scripts/android.py]: I've removed the comments.

File: scripts/android.py (hunk 2 above)
Comment (juan-lunarg) [reply scripts/android.py:46]: Or maybe the default should just be `arm64-v8a`... I'm not sure yet to be honest. This is still WIP

Comment (spencer-lunarg) [reply scripts/android.py:46]: I guess its like why we don't build both 32 and 64 by default for Linux/Windows... you should only pick what you need, one is enough for most developers who want to test on their local device

agree, should make sure we build 32-bit ARM in CI though

**Pull request merged**